frontend/public/module/k8s/k8s.ts

```diff
@@ -38,8 +38,61 @@ export const groupVersionFor = (apiVersion: string) => ({
   version: apiVersion.split('/').length === 2 ? apiVersion.split('/')[1] : apiVersion,
 });
 
+const parseAPIVersion = (version: string) => {
+  const parsed = /^v(\d+)(?:(alpha|beta)(\d+))?$/.exec(version);
+  return parsed
+    ? { majorVersion: Number(parsed[1]), qualifier: parsed[2], minorVersion: Number(parsed[3]) }
+    : null;
+};
+
+export const crdVersionSort = (v1: string, v2: string) => {
+  const v1Parsed = parseAPIVersion(v1);
+  const v2Parsed = parseAPIVersion(v2);
+
+  // Check null parsed versions first
+  if (_.isEmpty(v1Parsed) || _.isEmpty(v2Parsed)) {
+    // If a value fails null check order it last
+    return v1Parsed ? -1 : v2Parsed ? 1 : 0;
```

Comment:
Don't nest ternary operators. That gets confusing.

```suggestion
    if (v1Parsed) {
      return -1;
    }
    if (v2Parsed) {
      return 1;
    }
    return 0;
```

Reply:
Actually you should fall back to a localCompare if neither name is a valid version.

```suggestion
    if (v1Parsed) {
      return -1;
    }
    if (v2Parsed) {
      return 1;
    }
    return v1.localeCompare(v2);
```

Reply:
You beat me to it: https://github.com/openshift/console/pull/5355#discussion_r426004094